packages/ui/src/models/visualization/metadata/beans-entity-handler.test.ts

```diff
@@ -0,0 +1,150 @@
+import { BeansEntityHandler } from './beans-entity-handler';
+import { CamelRouteResource, KameletResource, PipeResource } from '../../camel';
+import * as kameletStub from '../../../stubs/kamelet-route';
+import * as routeStub from '../../../stubs/camel-route';
+import * as catalogIndex from '@kaoto-next/camel-catalog/index.json';
+import { CamelCatalogService } from '../flows';
+import { CatalogKind } from '../../catalog-kind';
+import { ICamelProcessorDefinition } from '../../camel-processors-catalog';
+
+describe('BeansEntityHandler', () => {
+  beforeAll(async () => {
+    const entitiesCatalog = await import('@kaoto-next/camel-catalog/' + catalogIndex.catalogs.entities.file);
+    /* eslint-disable  @typescript-eslint/no-explicit-any */
+    delete (entitiesCatalog as any).default;
+    CamelCatalogService.setCatalogKey(
+      CatalogKind.Entity,
+      entitiesCatalog as unknown as Record<string, ICamelProcessorDefinition>,
+    );
+  });
+
+  describe('should handle beans in CamelRouteResource', () => {
+    let model: any;
+    let beansHandler: BeansEntityHandler;
+    beforeEach(() => {
+      model = { ...routeStub.camelRouteJson };
```

Comment:
I think it's better to clone the `camelRouteJson` object, just in case we're performing modifications on it. We could use `cloneDeep` from lodash or [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone)

Reply:
ugh right, Done :+1: